blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,86 @@
+    category: Batch Changes
+  - description: "Code Insights drill-down filters now allow filtering insights data on the dashboard page using `repo:` filters."
+    url: https://github.com/sourcegraph/sourcegraph/issues/23186
+    category: Code Insights
+  - description: "Code Insights feature flag `DISABLE_CODE_INSIGHTS` environment variable has moved from the `repo-updater` service to the `worker` service. Any users of this flag will need to update their `worker` service configuration to continue using it."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23050
+    category: Code Insights
+  - description: "The Docker Compose Jaeger container's `SAMPLING_STRATEGIES_FILE` now has a default value. If you are using a custom sampling strategies configuration, you may need to make sure your configuration is not overridden by the change when upgrading."
+    url: "https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/489"
+    category: Admin
+  - description: The copy icon displayed next to files and repositories will now copy the file or repository path. Previously, this action copied the URL to clipboard.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23390
+    category: Search
+  - description: "For Docker-Compose and Kubernetes users, the built-in main Postgres and codeintel databases have switched to an alpine Docker image. This requires re-indexing the entire database. This process can take up to a few hours on systems with large datasets."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23697
+    category: Admin
+  - description: "All commit search results for `and`-expressions are now highlighted."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23336
+    category: Search
+  - description: Fixed a result streaming throttling issue that was causing significantly increased latency for some searches. This should improve the time to first rendered results for those searches.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23736
+    category: Search
+  - description: Updating draft merge requests on GitLab from batch changes no longer removes the draft status.
+    url: https://github.com/sourcegraph/sourcegraph/issues/23944
+    category: Batch Changes
+  
+---
+
+Sourcegraph 3.31 is now available! For this release, we introduced:
+
+## Code Insights is now in Private Beta
+
+Code Insights is the first tool ever that lets you to track the historical and future trends of anything in your codebase, from code migrations to API usage to linter overrides – or anything else you can run a Sourcegraph search for. Now in Private Beta, Code Insights can run across thousands of repositories, letting you then filter and explore your insights live. Reach out to your Account Executive for details.
+
+### Code Insights runs over all your repositories
+
+Code insights now runs over all repositories – we've removed the the ~50-70 repo limit.
```

Comment:
```suggestion
Code Insights now runs over all repositories – we've removed the the ~50-70 repo limit.
```

Reply:
@Joelkw @thisisvic - does this need to be a feature or can it be included in the changelog portion?